ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -51,6 +54,42 @@ var (
 	podUIDIndexField = "metadata.uid"
 )
 
+// getDiscoveryClient returns a discovery client for the current reconciler
+func getDiscoveryClient(config *rest.Config) (*discovery.DiscoveryClient, error) {
+	return discovery.NewDiscoveryClientForConfig(config)
+}
+
+// Check where we are running. We are trying to distinguish here whether
+// this is vanilla kubernetes cluster or OPenshift
+func getClusterType(logger logr.Logger) bool {
```

Comment:
How about using an environment variable for the KubeRay operator to determine which Kubernetes distribution is being used, instead of relying on automatic detection? Based on my experience, making assumptions about users' Kubernetes control planes can often lead to issues. In addition, environment variable will also make us easier to maintain the backward compatibility.

Reply:
e.g. `K8S_DISTRIBUTIONS`:

Case 1: it is not defined or `VANILLA` => vanilla Kubernetes
Case 2: `OPENSHIFT` => OpenShift

Reply:
You can refer to `ENABLE_INIT_CONTAINER_INJECTION` in #1069 for more details.

Reply:
Sorry, in my experience usage of env variable is by far more error-prone compared to automatic detection. Sorry

Reply:
I think I agree with automatic detection -- environment variables offer more control but tend to result in user-error more often.

Reply:
> I think I agree with automatic detection

https://github.com/ray-project/kuberay/pull/1185#pullrequestreview-1525067293

> environment variables offer more control but tend to result in user-error more often.

I believe only IBM/RedHat will use this feature, so we shouldn't subject other users to the risks of this code path.
In addition, there is no feature gate for this change.

Reply:
If you still want to use automatic detection, please add a feature flag. In addition, would you mind replying to my question (3) about the implementation in https://github.com/ray-project/kuberay/pull/1185#pullrequestreview-1525067293?

Reply:
```
I believe only IBM/RedHat will use this feature, so we shouldn't subject other users to the risks of this code path.
In addition, there is no feature gate for this change.
```
There are more users of OpenShift than IBM and RedHat, so it is relevant to all OpenShift users. We already have a flag bypassing this function. What exactly else do you want me to do?

Reply:
```
would you mind replying to my question (3) about the implementation in https://github.com/ray-project/kuberay/pull/1185#pullrequestreview-1525067293?
```
Yes, this is a standard way of using this check. See https://developers.redhat.com/blog/2020/09/11/5-tips-for-developing-kubernetes-operators-with-the-new-operator-sdk for an example